api/src/handlers/verify.rs

```diff
@@ -0,0 +1,212 @@
+use crate::handlers::process::{do_process_command, fetch_process_result};
+use crate::handlers::types::{ApiCommand, ApiCommandResult, VerifyResponse};
+use crate::rate_limiter::RateLimited;
+use crate::types::{ApiError, Result};
+use crate::utils::hardhat_config::HardhatConfigBuilder;
+use crate::utils::lib::{
+    check_file_ext, get_file_path, path_buf_to_string, status_code_to_message, ALLOWED_VERSIONS,
+    ARTIFACTS_ROOT, CARGO_MANIFEST_DIR, SOL_ROOT, ZK_CACHE_ROOT,
+};
+use crate::worker::WorkerEngine;
+use rocket::serde::json;
+use rocket::serde::json::Json;
+use rocket::tokio::fs;
+use rocket::State;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use tracing::info;
+use tracing::instrument;
+
+#[instrument]
+#[get("/verify/<version>/<network>/<contract_address>/<remix_file_path..>")]
+pub async fn verify(
+    version: String,
+    network: String,
+    contract_address: String,
+    remix_file_path: PathBuf,
+    _rate_limited: RateLimited,
+) -> Json<VerifyResponse> {
+    info!(
+        "/verify/{:?}/{:?}/{:?}/{:?}",
+        version, network, contract_address, remix_file_path
+    );
+    do_verify(version, network, contract_address, remix_file_path)
+        .await
+        .unwrap_or_else(|e| {
+            Json(VerifyResponse {
+                message: e.to_string(),
+                status: "Error".to_string(),
+            })
+        })
+}
+
+#[instrument]
+#[get("/verify-async/<version>/<network>/<contract_address>/<remix_file_path..>")]
```

Comment:
Maybe we should add validation of "contract_address"

Reply:
In case a malformed address is provided, hardhat will spit out a good enough error that should put the user back on track.

If you still want to have the validation in, is there any resource you can point me to show how to do it? Not sure what a well-formatted address must look like.